### PR TITLE
Update Django admin views for bulk moderation ergonomics

### LINKED ISSUE #3842: Update admin DeletedMedia view with bulk actions and filters
## Description

Blocked by #3840

* Add the action for bulk reverse deindexing (implemented in #3840) to the DeletedMedia view
* Add a new filter to the DeletedMedia view for filtering by ModerationDecision id

## Additional context

See [IP](https://docs.openverse.org/projects/proposals/trust_and_safety/content_report_moderation/20240122-implementation_plan_bulk_moderation_actions.html#deletedmedia-list-view) for context.

### LINKED ISSUE #3843: Update admin SensitiveMedia view with bulk actions and filters
## Description

Blocked by #3840

Update the SensitiveMedia view in Django admin to:

* Add the action for bulk reverse mark sensitive (implemented in #3840) to the SensitiveMedia view
* Add a new filter to the SensitiveMedia view for ModerationDecision id

## Additional context

See [IP](https://docs.openverse.org/projects/proposals/trust_and_safety/content_report_moderation/20240122-implementation_plan_bulk_moderation_actions.html#sensitivemedia-list-view) for more details.

### LINKED ISSUE #3844: Create the ModerationDecision admin views
## Description

Blocked by #3636

Create a Django admin ListView for the ModerationDecision model, for viewing historical ModerationDecisions. In the list view we should sort decisions by creation date and display:

* `id`
* `action`
* `explanation` (truncated)
* `record_count` - a count for the number of records affected by the decision

The view should have a custom filter to display only bulk moderation decisions, defined as a ModerationDecision with more than one media record.

Also include a **read-only** detail view for individual ModerationDecisions when selected from the list. For ModerationDecisions with a `deindexed_sensitive`, `deindexed_copyright`, or `marked_sensitive` action type, include a link to the SensitiveMedia or DeletedMedia view, filtered by the ModerationDecision id for convenience.

## Additional context

See [IP](https://docs.openverse.org/projects/proposals/trust_and_safety/content_report_moderation/20240122-implementation_plan_bulk_moderation_actions.html#moderationdecisions-list-view) for details.

### PR DESCRIPTION
**Pull request by dhruvkb**
## Fixes

Fixes #3842 by @stacimc 
Fixes #3843 by @stacimc
Fixes #3844 by @stacimc 

## Description

1. This PR adds a filter by decision ID to the sidebar of the sensitive media and deleted media list views. Using this filter the list can be quickly narrowed down to all the meida affected by one decision, making it easier to operate on them again, as a bulk.

   image

2. This PR also updates the decision list view to add a single/bulk decision filter.

   image

3. Finally the PR links to the sensitive media or deleted media list views from the single decision page, automatically populating the filter parameter described in point 1 above.

  Screenshot 2024-07-30 at 9 54 55 PM

## Testing Instructions

0. Check out the PR and run the dev server.
1. Create some reports and take decisions on them from the single media view or take decisions on media items directly using bulk moderation actions from the media list view.
2. Go to the sensitive media list and filter by a decision ID. You should only UUIDs for  media that were affected by that decision.
3. Go to the deleted media list and filter by a decision ID. You should only see UUIDs for media that were affected by that decision.
4. Go the decision list view. Filter the decisions by single or bulk. Based on your actions in step 1, you will see a subset of decisions.
5. Click on a deindexing decision, you will see a "View deleted" button in top right, clicking which takes you to the pre-filtered deleted media list view.
6. Click on a sensitive marking decision, you will see a "View sensitive" button in the top right, clicking which tkaes you to the pre-filtered sensitive media list view.
 
## Checklist

- [x] My pull request has a descriptive title (not a vague title like`Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.
- [ ] I ran the DAG documentation generator (`ov just catalog/generate-docs` for catalog
      PRs) or the media properties generator (`ov just catalog/generate-docs media-props`
      for the catalog or `ov just api/generate-docs` for the API) where applicable.

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: api/api/templates/admin/api/media_decision/change_form.html [hunk 1]
```diff
@@ -0,0 +1,15 @@
+{% extends "admin/change_form.html" %}
+
+{% block object-tools-items %}{{ block.super }}
+<li>
+  {% if decision_obj.action == "deindexed_copyright" or decision_obj.action == "deindexed_sensitive" %}
+  <a href="{% url 'admin:api_deleted'|add:media_type|add:'_changelist' %}?decision_id={{ object_id}}">
+    View deleted
```
Comment (zackkrida) [inline api/api/templates/admin/api/media_decision/change_form.html:7]:
```suggestion
    View deleted media
```

File: api/api/templates/admin/api/media_decision/change_form.html [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  </a>
+  {% elif decision_obj.action == "marked_sensitive" %}
+  <a href="{% url 'admin:api_sensitive'|add:media_type|add:'_changelist' %}?decision_id={{ object_id}}">
+    View sensitive
```
Comment (zackkrida) [inline api/api/templates/admin/api/media_decision/change_form.html:11]:
```suggestion
    View sensitive media
```

File: api/api/templates/admin/api/media_decision/change_form.html (hunk 1 above)
Comment (dhruvkb) [reply api/api/templates/admin/api/media_decision/change_form.html]: Applied in dfc9918.

File: api/api/templates/admin/api/media_decision/change_form.html (hunk 2 above)
Comment (dhruvkb) [reply api/api/templates/admin/api/media_decision/change_form.html]: Applied in dfc9918.

File: api/api/admin/media_report.py [hunk 3]
```diff
@@ -857,6 +964,23 @@ def media_ids(self, obj):
     # Change view #
     ###############
 
+    change_form_template = "admin/api/media_decision/change_form.html"
+
+    def change_view(self, request, object_id, form_url="", extra_context=None):
+        # Expand the context based on the template's needs.
```
Comment (zackkrida) [inline api/api/admin/media_report.py:970]:
```suggestion
```

Nit: this comment seems somewhat redundant or self-explanatory.

File: api/api/admin/media_report.py [hunk 4]
```diff
@@ -222,6 +222,110 @@ def queryset(self, request, qs):
...
+                return queryset
+
+            queryset = queryset.annotate(
+                media_count=Count(f"{media_type}decisionthrough")
+            )
+
+            if self.value() == "single":
+                return queryset.filter(media_count=1)
+            elif self.value() == "bulk":
+                return queryset.filter(media_count__gt=1)
```
Comment (zackkrida) [inline api/api/admin/media_report.py:324]:
```suggestion
        def queryset(self, request, queryset):
    				filter_conditions = {
        			  "single": {"media_count": 1},
        			  "bulk": {"media_count__gt": 1},
    				}
    				
            if self.value() not in filter_conditions:
                return queryset

            queryset = queryset.annotate(
                media_count=Count(f"{media_type}decisionthrough")
            )

    				return queryset.filter(**filter_conditions[self.value()])
```

More of a question than a nit: would this be a bit more "pythonic"? It seems like a pattern I've come across in my limited Python experience. I'm genuinely curious though, this is less of a suggestion and more of a personal curiosity.

**Review by zackkrida — APPROVED** [review]
Works great! Minor comments (you've already addressed one of my typical comments favoring verbosity over potentially-ambiguous brevity), but LGTM.

File: api/api/admin/media_report.py (hunk 3 above)
Comment (dhruvkb) [reply api/api/admin/media_report.py:970]: It's been used in one more place so I added it for consistency. It also explains why we had to override the entire view just to add some more context for the template.

File: api/api/admin/media_report.py (hunk 4 above)
Comment (dhruvkb) [reply api/api/admin/media_report.py]: I don't have a preference for either of the two, and yours seems much cleaner. So I updated it in 45837f0.

**Pull request merged**